cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -430,10 +431,57 @@ func (scaleSet *ScaleSet) buildNodeFromTemplate(template compute.VirtualMachineS
 
 	// GenericLabels
 	node.Labels = cloudprovider.JoinStringMaps(node.Labels, buildGenericLabels(template, nodeName))
+	// Labels from the Scale Set's Tags
+	node.Labels = cloudprovider.JoinStringMaps(node.Labels, extractLabelsFromScaleSet(template.Tags))
+
+	// Taints from the Scale Set's Tags
+	node.Spec.Taints = extractTaintsFromScaleSet(template.Tags)
+
 	node.Status.Conditions = cloudprovider.BuildReadyConditions()
 	return &node, nil
 }
 
+func extractLabelsFromScaleSet(tags map[string]*string) map[string]string {
```

Comment:
These are similar methods to the way the AWS cloud provider handles the same scenario, I left attempting to refactor the AWS and Azure cloud providers to share code outside of the scope of this PR